### DISCUSSION TITLE
understanding individual postanalysis results_Preprocess Table

### DISCUSSION THREAD

**Discussion by singhd6** [milaboratory/mixcr#1876] [Q&A] [answered]
**Hi I want to understand the **preprocess table**, my command to generate the table was**: 
![image](https://github.com/user-attachments/assets/4ee273db-f825-4cd6-accb-abd7d17fdc2f)
[individual_result.preprocessTable_DS.xlsx](https://github.com/user-attachments/files/18282338/individual_result.preprocessTable_DS.xlsx)
According to my knowledge and according to mixcr documentation the explanation for each column is: 
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

characteristic:  name for a group of metrics for which the   same downsampling was applied
--
preprocessor: the name of the   overall preprocessing chain applied to the dataset
nElementsBefore:   number of clonotypes before any preprocessing applied (that is in the initial   dataset)
sumWeightBefore:   sumWeightBefore
nElementsAfter: number of   clonotypes in the dataset after all preprocessing applied
sumWeightBefore: total weight   of all clonotypes before any preprocessing applied
nElementsAfter: number of   clonotypes in the dataset after all preprocessing applied
preprocessor#i and following columns: report columns for i-th step in   the preprocessing chain

</body>

</html>
![image](https://github.com/user-attachments/assets/25248234-f5b8-4054-a550-efce7a982001)
What does each column represent: 
<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

nElementsBefore
--

</body>

</html>

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

sumWeightBefore
--

</body>

</html>

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

nElementsAfter
--

</body>

</html>

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

sumWeightAfter
--

</body>

</html>

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

preprocessor#1
--

</body>

</html>

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

nElementsBefore#1
--

</body>

</html>

<html xmlns:o="urn:schemas-microsoft-com:office:office"

<html xmlns:o="urn:schemas-microsoft-com:office:office"
xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

nElementsAfter#1
--

</body>

</html>

xmlns:x="urn:schemas-microsoft-com:office:excel"
xmlns="http://www.w3.org/TR/REC-html40">

<head>

<meta name=ProgId content=Excel.Sheet>
<meta name=Generator content="Microsoft Excel 15">
<link id=Main-File rel=Main-File
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip.htm">
<link rel=File-List
href="file:///C:/Users/singhd6/AppData/Local/Temp/msohtmlclip1/01/clip_filelist.xml">
<style>

</style>
</head>

<body link="#467886" vlink="#96607D">

sumWeightBefore#1
--

</body>

</html>

sumWeightAfter#1
preprocessor#2	
nElementsBefore#2	
sumWeightBefore#2	
nElementsAfter#2	
sumWeightAfter#2	
preprocessor#3

![image](https://github.com/user-attachments/assets/ed84bde4-91fb-4692-8873-c1e200fb0c2d)

And then in the final frequency table how are we deciding the frequencies or fraction of v gene usage? Is it the number of total clones divided by the number of clones used by v gene?
I understand the explanation of gene frequencies but where did it come from?
frequencies for each v gene/segment present in the sample, during the creation of plot the weight or numbers represent the fraction of clones with a certain v gene regardless of their abundance in the repertoire.

**Comment by mizraelson** [comment, marked as answer]
Since you did not use any weight function this is a fraction of clones with a certain V gene. The number of clones with a certain V gene divided by the total number of clones.

The columns in the preprocess tables represent the number of clones (nElements) and the number of read (sumWeight) before and after each per-processor was applied.

**Reply by singhd6** [reply]
Thank you very much. 
Sincerely,
Divya
